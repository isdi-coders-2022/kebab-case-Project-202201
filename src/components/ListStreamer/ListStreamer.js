import FavoriteStreamer from "../FavoriteStreamer/FavoriteStreamer";
import styled from "styled-components";
import propTypes from "prop-types";

const Container = styled.div`
  background-color: #eaeae1;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
`;

const ListStreamer = ({ streamers }) => {
  return (
    <Container className="container">
      {streamers.map((streamer) => {
        return <FavoriteStreamer key={streamer} streamerInfo={{ streamer }} />;
      })}
    </Container>
  );
};

ListStreamer.prototype = {
  streamerInfo: propTypes.object,
};

export default ListStreamer;
