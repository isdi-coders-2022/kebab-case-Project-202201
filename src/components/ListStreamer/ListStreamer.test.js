import { screen, render } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import ListStreamer from "./ListStreamer";

describe("Given a ListStreamer", () => {
  let streamers = null;
  beforeAll(() => {
    streamers = [1];
  });
  describe("When called a list of streamers", () => {
    test("It should display a list of streamers", () => {
      render(<ListStreamer streamers={streamers} />);

      const renderedElement = screen.getByRole("article");

      expect(renderedElement).toBeInTheDocument();
    });
  });
  describe("When it get a const App that creates Header", () => {
    test("then it should create and compare a snapshot with Header component", () => {
      const app = TestRenderer.create(<ListStreamer streamers={streamers} />);
      expect(app.toJSON()).toMatchSnapshot();
    });
  });
});
