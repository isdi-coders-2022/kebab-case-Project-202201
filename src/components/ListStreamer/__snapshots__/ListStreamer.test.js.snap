// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ListStreamer When it get a const App that creates Header then it should create and compare a snapshot with Header component 1`] = `
<div
  className="sc-dkPtRN eElyIp container"
>
  <article
    className="sc-bdvvtL cforNZ"
  >
    <header
      className="sc-gsDKAQ lplKdE"
    >
      <img
        alt="undefined"
      />
      <div>
        <h2 />
      </div>
      <button>
        shish-kebab
      </button>
    </header>
    <div>
      Reel Container
    </div>
  </article>
</div>
`;
